src/sysusers/sysusers.c

```diff
@@ -534,6 +534,10 @@ static int write_temporary_shadow(const char *shadow_path, FILE **tmpfile, char
                         }
 
                         errno = 0;
+                        /* Make sure we keep the NIS entries (if any) at the end. */
+                        if (IN_SET(sp->sp_namp[0], '+', '-'))
+                                break;
+
```

Comment:
please set errno = 0 right before the putspent() btw, i.e. swap this if check and the errno initialization please